msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/ManagedIdentityTests.java

```diff
@@ -0,0 +1,323 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.net.SocketException;
+import java.net.URISyntaxException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_METHOD)
+public class ManagedIdentityTests {
```

Comment:
Can you add a test that checks the token cache keys if one doesn't exist? We should make sure that AT key is consistent with .NET for SAMI and for UAMI. I understand it's only in memory.

Reply:
There is a difference in tenant id that is used in Java to build the cache key vs the tenant id used in .Net. Other than that the cache key is same. Will need some more work to make them same. Something I think need to be consistent with .Net https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/src/client/Microsoft.Identity.Client/Instance/Authority.cs#L45, rules to set the tenant id.

Reply:
Wil create a task on board for this.